src/settings/MappingProfiles/MappingProfilesTransformationsModal/MappingProfilesTransformationsModal.js

```diff
@@ -70,10 +82,13 @@ export const MappingProfilesTransformationsModal = ({
   let displayedCheckedItemsAmount = 0;
 
   searchValueResults.forEach(transformation => {
-    if (get(searchFilters, 'recordTypes', []).includes(transformation.recordType)) {
+    if (
+      get(searchFilters, 'recordTypes', []).includes(transformation.recordType)
```

Comment:
I think it is overcomplicated and you can try to split it. For example, introducing intermediate variables

Reply:
I thought about this but in this case, the checks will occur for both record types and statuses. With the current approach, if the check for record types is false no check will occur for statuses.

Reply:
I agree with you. But the current solutions looks too complex for reading. 
You can try:
- move  get(searchFilters, 'recordTypes', [])  and  get(searchFilters, 'status', []) to variables. This is not expensive to get field from the object
- move the first check to the good named variable because this code will be executed in any case
- your options

What do you think about it, Viktor?

Reply:
This is possible. I also thinking about a more generic way to handle this. Please take a look and share your feedback.

```
  const filterMap = {
    recordTypes: (filters, transformation) => filters.includes(transformation.recordType),
    statuses: (filters, transformation) => filters.some(status => statusesFilterMap[status](transformation, selectedTransformations)),
  };

  searchValueResults.forEach(transformation => {
    const hasFilterMatch = Object.keys(filterMap)
      .some(filterKey => {
        const filter = get(searchFilters, filterKey, []);

        return filterMap[filterKey](filter, transformation);
      });

    if (hasFilterMatch) {
      searchResults.push(transformation);

      if (selectedTransformations[transformation.fieldId]) {
        displayedCheckedItemsAmount++;
      }
    }
  });
```

Reply:
Looks ok